### ISSUE TITLE
Rocket Bow Crash

### ISSUE THREAD

**Issue by SoulBlast35** [nosoop/SM-TFAttributeSupport#3]
This isnt a bug with the plugin, but it's something that could possibly be fixed by the plugin? Setting a Sniper Bow to fire rockets causes the server/game to hard crash upon firing a second rocket (possibly while the first rocket is still active? Unsure). Other projectiles work completely fine, such as Flares, Mangler Blasts and Bison Lasers. This bug also affects robot Bowmen snipers in MvM, causing them to crash the game upon firing a second rocket. This bug exists on Listen servers too, so it is not a server plugin issue. Is it possible to fix this?
EDIT: Jarate (and by extension, Mad Milk) appear to be VERY unstable on Bows. Sometimes it will work, sometimes it will hard crash.
EDIT2: Okay, I THINK it is something to do with tap-firing the Bow; Aka not charging the bow. Sniper-bot AI sometimes does this if youre close to them.

**Comment by nosoop** [comment]
Yep, managed to reproduce it on just tap-firing.  Inspecting the crash dump suggests that some particular interaction here isn't handled very well in the scripted scene system.

This would definitely be something that could use fixing.  I have an idea for a hotfix; will test it out real quick.

**Comment by nosoop** [comment]
The stack trace was misleading; not actually in the scene system, whatever bug this is seems to just cause some subtle issues later on.

Have tried:
- Preventing `CTFPlayer::SpeakWeaponFire()` from calling into the response system
- Returning `null` on `CTFWeaponBase::FireProjectile()` so `CTFCompoundBow::LaunchGrenade()` doesn't get a `CTFProjectile_Arrow` instance back.

**Comment by nosoop** [comment]
It's also happening on fully charged shots.  Blocking `CTFWeaponBase::FireProjectile()` in its entirety prevents the crash, but obviously prevents the projectile from spawning as well.

Unfortunately the stack trace doesn't have a clear indicator of what's up (corruption in the stack itself, maybe?), so spending the time doing research on this problem is going to be low priority.
